packages/babel-plugin-extract-react-types/index.js

```diff
@@ -0,0 +1,37 @@
+const { findExportedComponents } = require('extract-react-types');
+
+module.exports = babel => {
+  let t = babel.types;
+  return {
+    visitor: {
+      Program(programPath, state) {
+        let typeSystem = state.file.opts.parserOpts.plugins
+          .map(plugin => (Array.isArray(plugin) ? plugin[0] : plugin))
+          .find(plugin => plugin === 'flow' || plugin === 'typescript');
```

Comment:
Just trying to understand this correctly. Where is the plugin list derived from? Is it file specific?

Example problem is atlaskit contains both flow and typescript files. If both plugins exist, do we just get the first one?

Reply:
It's comes from the syntax plugins which are included in the flow and typescript presets. Both plugins can't exist together. I just checked atlaskit's webpack and babel configs and babel is only used on untyped/flow code whereas the typescript compiler is used on ts files so this babel plugin wouldn't work for ts unless you switched from the ts compiler to babel (at least on the website). To do that, you would probably want to use babel's overrides option and use the flow plugin on .js files and the typescript plugin on .ts and .tsx files.